include/msgpack/sysdep.h

```diff
@@ -36,7 +36,7 @@ typedef unsigned __int64 uint64_t;
 #include <stdbool.h>
 #endif
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(_LIB)
```

Comment:
This is reasonable change to support static-lib on MSVC. However, it is a different topic. Could you separate it from the pull request?